$OpenBSD: patch-array_c,v 1.1 2018/04/10 22:51:28 kn Exp $

Fix segfault on repeated pattern searching, cherry-picked
7d1f70e18cce00ca3fea43f392a3ea3a367f18b9.

Index: array.c
--- array.c.orig
+++ array.c
@@ -119,7 +119,7 @@ bool array_remove(Array *arr, size_t idx) {
 	}
 	char *dest = arr->items + idx * arr->elem_size;
 	char *src = arr->items + (idx + 1) * arr->elem_size;
-	memmove(dest, src, (arr->len - idx) * arr->elem_size);
+	memmove(dest, src, (arr->len - idx - 1) * arr->elem_size);
 	arr->len--;
 	return true;
 }
